demes/demes.py

```diff
@@ -1412,78 +1429,62 @@ def deme(
                     f"of existence for ancestor {ancestor} "
                     f"({anc.start_time}, {anc.end_time})"
                 )
-
-        # build the deme's epochs
-        if epochs is None:
-            # if epochs are not given, we assign a single epoch over that deme
-            if final_size is None:
-                final_size = initial_size
-            if end_time is None:
-                end_time = 0
-            if initial_size == final_size:
-                size_function = "constant"
-            else:
-                size_function = "exponential"
-            epochs = [
-                Epoch(
-                    start_time=start_time,
-                    end_time=end_time,
-                    initial_size=initial_size,
-                    final_size=final_size,
-                    size_function=size_function,
-                    selfing_rate=selfing_rate,
-                    cloning_rate=cloning_rate,
-                )
-            ]
+        # set default cloning and selfing rates
+        if "selfing_rate" in defaults:
+            selfing_rate = defaults["selfing_rate"]
+        elif "selfing_rate" in self.defaults:
+            selfing_rate = self.defaults["selfing_rate"]
         else:
-            # the last epoch needs to have an end time
-            if epochs[-1].end_time is None:
-                if end_time is not None:
-                    epochs[-1].end_time = end_time
+            selfing_rate = None
+        if "cloning_rate" in defaults:
+            cloning_rate = defaults["cloning_rate"]
+        elif "cloning_rate" in self.defaults:
+            cloning_rate = self.defaults["cloning_rate"]
+        else:
+            cloning_rate = None
```

Comment:
How about:
```suggestion
        cloning_rate = defaults.get("cloning_rate", self.defaults.get("cloning_rate"))
```
And similarly for `selfing_rate` above.